zgw/notificaties-api/src/main/kotlin/com/ritense/notificatiesapi/NotificatiesApiPlugin.kt

```diff
@@ -16,19 +16,105 @@
 
 package com.ritense.notificatiesapi
 
+import com.ritense.notificatiesapi.client.NotificatiesApiClient
+import com.ritense.notificatiesapi.domain.Abonnement
+import com.ritense.notificatiesapi.domain.Kanaal
+import com.ritense.notificatiesapi.domain.NotificatiesApiAbonnementLink
+import com.ritense.notificatiesapi.domain.NotificatiesApiConfigurationId
+import com.ritense.notificatiesapi.repository.NotificatiesApiAbonnementLinkRepository
 import com.ritense.plugin.annotation.Plugin
 import com.ritense.plugin.annotation.PluginProperty
+import com.ritense.plugin.domain.PluginConfigurationId
 import java.net.URI
+import java.security.SecureRandom
+import java.util.Base64
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import mu.KotlinLogging
 
 @Plugin(
     key = "notificatiesapi",
     title = "Notificaties API",
     description = "Enable interfacing with Notificaties API specification compliant APIs"
 )
-class NotificatiesApiPlugin {
+class NotificatiesApiPlugin(
+    private val client: NotificatiesApiClient,
+    private val abonnementLinkRepository: NotificatiesApiAbonnementLinkRepository
+) {
     @PluginProperty(key = "url", secret = false)
     lateinit var url: URI
 
     @PluginProperty(key = "authenticationPluginConfiguration", secret = false)
     lateinit var authenticationPluginConfiguration: NotificatiesApiAuthentication
+
+    fun createAbonnement(
+        pluginConfigurationId: PluginConfigurationId,
```

Comment:
Since we're already within the context of a plugin instance, maybe it makes more sense to provide the plugin configuration id another way, e.g. through the pluginfactory.

Reply:
Also fair thing to consider. I do wonder how much that adds to this specific plugin, given that it's not meant to do much on it's own and is mainly a plugin that other plugins/services depend on.

Thoughts? @Klaas-Ritense